prover/src/lib.rs

```diff
@@ -107,10 +118,14 @@ impl ExecutionProver {
     }
 }
 
-impl Prover for ExecutionProver {
+impl<H> Prover for ExecutionProver<H>
+where
+    H: Clone + Hasher,
+{
     type Air = ProcessorAir;
     type BaseField = Felt;
     type Trace = ExecutionTrace;
+    // TODO should be H
```

Comment:
these TODOs should probably have its own issue if not done on this PR, no?

Reply:
Agreed! I usually leave these questions to be checked @ review and create the issues after discussion/before merging